SpikesBot.js

```diff
@@ -0,0 +1,333 @@
+            if (guildChannel.name === 'general') {
+
+                //add it to the list
+                generalChannelsForTeamTime[generalChannelsForTeamTime.length] = guildChannel
+            }
+        })
+    });
+}
+
+//calc when to send the TEAMTIME message
+function whenIsTeamTime() {
+    const now = new Date() //the current time
+
+    //calc the remaining time until the teamtime in milliseconds
+    let remainingTime = (TEAMTIME.getHours() - now.getHours()) * 60 * 60 * 1000
+    remainingTime += (TEAMTIME.getMinutes() - now.getMinutes()) * 60 * 1000
+    remainingTime -= now.getSeconds() * 1000
+
+    if (remainingTime >= 0) {
+    } else {
+        remainingTime += 24 * 60 * 60 * 1000
+    }
+
+    //return the time until the next teamtime
+    console.log("next TEAMTIME: " + remainingTime);
+    return remainingTime;
+}
+
+function getSpecialCommands(){
+    let allCommands = ''
+    for(const someCommand in SPECIAL_COMMANDS){
+        allCommands += '||**' + someCommand + '**||, '
+    }
+    return allCommands;
+}
+
+//fetch info about a team from the blue alliance website
+function fetchTeamLinks(someTeamNumber) {
+
+    return fetch('https://www.thebluealliance.com/team/' + someTeamNumber)
+        .then(response => response.text())
+        .then(htmlContent => {
+            //get the html area of the team's social media accounts
+            let social = htmlContent.split('team-social-media')[1].split('</div>')[0]
+            let links = social.split('href="')
+
+            //for each link
+            for (let i = 0; i < links.length; i++) {
+                //cut to get only the links
+                links[i] = links[i].split('"')[0]
+            }
+            links.splice(0, 2); //get rid of non links
+            return links;
+        })
+        .catch(error => {
+            return ['fetch error occurred, please try again later'];
+        });
+}
+
+//search for a specific social media of a specific team
+function searchSocialMedia(searchItem, teamNumber) {
+
+    try {
+        //fetch all of the social media links of a team
+        return fetchTeamLinks(teamNumber)
+            .then(links => {
+
+                //in each link
+                for (const link of links) {
+
+                    //search for the specific social media
+                    if (link.includes(searchItem)) {
+                        return link;
+                    }
+                }
+                return 'not found';
+            });
+    } catch {
+        return 'not found';
+    }
+
+}
+
+client.on('messageCreate', msg => {
+
+    const everyoneRole = msg.guild.roles.everyone;
+
+    // Check if the @everyone role has the `Send Messages` permission in the channel
+    const permissions = msg.channel.permissionsFor(everyoneRole);
+
+    //check that the message sent by a user and @everyone can send messages in the channel
+    if (!msg.author.bot && permissions.has(PermissionsBitField.Flags.SendMessages)) {
```

Comment:
add a check to make sure reaction permission is also enabled since it is used by the SpikesBot and could also make the bot crash